std/math.d

```diff
@@ -7760,9 +7760,9 @@ if (isFloatingPoint!(F) && isIntegral!(G))
     {
         switch (n)
         {
-        case -1:
+        case cast(G) -1:
             return 1 / x;
-        case -2:
+        case cast(G) -2:
             return 1 / (x * x);
```

Comment:
I think it would be better to write this as 
```
 if (isSigned!G && n < 0)
 {
         switch (cast(SignedOf!G)n)
```

That way the meaning of the label remain uncluttered.

Reply:
That SignedOf thing does not appear to exist or be public to me

Reply:
I thought there would be something like that in `std.traits`

Reply:
I've used another approach which is to promote the switch condition and delegate the task to std.conv.to